tools/abel/README.md

```diff
@@ -0,0 +1,164 @@
+
+    ```
+        curl -X POST \
+        -H "Authorization: Bearer "$(gcloud auth application-default print-access-token) \
+        -H "Content-Type: application/json; charset=utf-8" \
+        -d @token_request.json \
+        "https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/your-service-account@project-id.iam.gserviceaccount.com:generateAccessToken" >| spreadsheet_token.json
+    ```
+
+4. Confirm `spreadsheet_token.json` looks something like:
+
+    ```
+    {
+      "accessToken": "ya29.c.AXv0zTOa9p5K78GtGntoIQCqyt2R572qZ3…",
+      "expireTime": "2022-02-24T19:57:26Z"
+    }
+    ```
+
+ ***note:*** Spreadsheet tokens only last one hour and the command from step #3
+ must be run again to generate a new token.
+
+## Using ABEL
+ABEL has a few pieces of core functionality, they are:
+* Ingest an [ABEL spreadsheet](../../tools/abel/validators/README.md) and export a valid [Building Config](../../ontology/docs/building_config.md) file
+* Ingest a [Building Config](../../ontology/docs/building_config.md) and export an [ABEL spreadsheet](../../tools/abel/validators/README.md)
+
+### Command-line arguments for ABEL:
+`-s` or `--spreadsheet_id` **required** id for a google sheets spreadsheet
+  * [ABEL Spreadsheet Template](https://docs.google.com/spreadsheets/d/1qKMlpJI5-_h_8innNniEkpatMBcRHSGekrRwTsPQ618/copy#gid=980240783)
+  * A Google Sheets ID is found embedded into the spreadsheet's url.
+  e.g. `https://docs/google/com/spreadsheets/d/<spreadsheet_id>/edit#gid=123467`
+
+`-b` or `--building_config` absolute path to a local building configuration file
+  * [Building Configuration Docs](../../ontology/docs/building_config.md)
+
+`-t` or `--token` path to the GCP project token. Default path for
+  the token is the current directory, but an alternate relative or absolute path
+  may be provided.
+
+### The ABEL Spreadsheet
+The ABEL spreadsheet serves as a user-friendly interface for ABEL and is what
+allows a user to make changes to machine readable documents like [Building
+Config](../../ontology/docs/building_config.md).
+
+Please see the [ABEL spreadsheet docs](../../tools/abel/validators/README.md) for detailed instructions on how to create your own spreadsheet.
+
+### Spreadsheet -> Building Config
+
+The process for using an ABEL spreadsheet to generate a Building Config is as
+follows:
+
+1. Create a spreadsheet for ABEL from [ABEL Spreadsheet template](https://docs.google.com/spreadsheets/d/1tcLjFnHiXUT-xh5C1hRKiUVaUH_CzgSI8zFQ_B8q7vs/copy#gid=980240783)
+2. Share your spreadsheet with your GCP service account and project id as an editor. Refer to Google Sheets documentation on [how to share a google sheet](https://support.google.com/docs/answer/9331169?hl=en#6.1)
+3. Populate your spreadsheet. A well defined guide on how to populate your
+   spreadsheet can be found in the [spreadsheet docs](../../tools/abel/validators/README.md)
+4. Run ABEL with the command:
+```
+python3 abel.py -s <input_spreadsheet_id>
```

Comment:
token needed?

Reply:
Token is not needed, I added a more detailed note in the command-line arguments section.